remoter-server/src/cellSizeDistribution.r

```diff
@@ -0,0 +1,99 @@
+#'          to call generate_default_values_cellSizeDistribution)
+#'          - filterSettings: slot with thresholds
+#'                  - minCellSize: Integer. Threshold that contain the minimun number of UMIs per cell
+#'                  - binStep: Integer. Bin size for the histogram
+#' @export return a list with the filtered seurat object by cell size ditribution, the config and the plot values
+
+
+# CalculateBarcodeInflections calculates an adaptive inflection point ("knee")
+# of the barcode distribution for each sample group. This is
+# useful for determining a threshold for removing low-quality
+# samples.
+generate_default_values_cellSizeDistribution <- function(seurat_obj, config) {
+  import::here(CalculateBarcodeInflections, .from = Seurat)
+  import::here(Tool, .from = Seurat)
+  # Q: should we check for precalculated values? e.g.:
+  # is.null(Tool(seurat_obj, slot = "CalculateBarcodeInflections"))
+  # Returns Seurat object with a new list in the `tools` slot,
+  # `CalculateBarcodeInflections` including inflection point calculatio
+  seurat_obj_tmp <- CalculateBarcodeInflections(
+                                        object = seurat_obj,
+                                        barcode.column = "nCount_RNA",
+                                        group.column = "orig.ident",
+                                        # [HARDCODED]
+                                        threshold.low = 1e2,
+                                        threshold.high = NULL
+  )
+  # extracting the inflection point value which can serve as minCellSize threshold
+  # all other side effects to the scdate object will be discarded
+  # [TODO unittest]: this calculation assumes a single sample, i.e. only one group in orig.ident!
+  # otherwise it will return multiple values for each group!
+  # returned is both the rank(s) as well as inflection point
+  # orig.ident          nCount_RNA  rank
+  # SeuratProject       1106        10722
+  tmp <- Tool(seurat_obj_tmp, slot = "CalculateBarcodeInflections")$inflection_points
+  # extracting only inflection point(s)
+  return(tmp$nCount_RNA)
+}
+
+task <- function(seurat_obj, config) {
+    import::here(map2, .from = purrr)
+    minCellSize <- as.numeric(config$filterSettings["minCellSize"])
+    # extract plotting data of original data to return to plot slot later
+    plot1_data  <- seurat_obj$nCount_RNA
+    # plot 1: histgram of UMIs, hence input is all UMI values, e.g.
+    # AAACCCAAGCGCCCAT-1 AAACCCAAGGTTCCGC-1 AAACCCACAGAGTTGG-1
+    #               2204              20090               5884  ...
+    #names(plot1_data) <- rep("u", length(plot1_data))
+   
+    #    u    u    u    u    u    u
+    # 3483 6019 3892 3729 4734 3244
+    plot2_data <- order(seurat_obj$nCount_RNA)
+    #names(plot2_data) <- rep("rank", length(plot2_data))
+    plot2_data <- unname(map2(plot1_data,plot2_data,function(x,y){c("u"=x,"rank"=y)}))
+    plot1_data <- lapply(unname(plot1_data),function(x) {c("u"=x)})
```

Comment:
We need to check if it actually works with the UI, most likely it will work, just need to finally test and then apply this to all other filters.